### DISCUSSION TITLE
Updating metadata in photoprism when photos are updated

### DISCUSSION THREAD

**Discussion by jonfk** [photoprism/photoprism#3207] [Q&A] [answered]
Hi,
I am a new photoprism user. I imported my photo library but noticed that the date taken metadata on some of my photos were incorrect. Since I can see that photoprism doesn't update the original metadata on the file when updated in photoprism, I used lightroom to update the files directly. I prefer this since I would like to use these files with more programs than photoprism.

Once updated, I rescanned the photos but this doesn't seem to update them in photoprism (i.e. they are still showing the old date). I tried to rescan both with and without `Complete Rescan` checked but I doesn't seem to make a difference. What would be the expected way to have them updated in photoprism? 

I tried to find a way to remove the photos from photoprism and rescan them but I can't seem to find a way. The archive feature doesn't remove the photos and the delete feature seems to delete them from disk permanently. 

For now the only solution I can find would be to delete my photoprism library and restart from scratch but I don't see that as a good solution. What if I find more photos that need updated metadata?

Is there a better way or is this not currently supported in photoprism?

**Comment by graciousgrey** [comment, marked as answer]
If you update the metadata of your original files, PhotoPrism can read these updates during the next indexing, given the [metadata field](https://www.photoprism.app/kb/metadata) you edited is supported and given that the tool you use to update the metadata also updates the files modify date.

**Reply by jonfk** [reply]
I see. It is a supported feature then. Could this be a bug or am I doing something wrong? 

The field is the `Date Taken` field of the exif. It seems to be supported since photoprism picked up the date originally in that field when I first indexed it. But it doesn't show the updated date even after re-indexing. All other programs that I am using to read the file show the new date that was written. 

Is there anything I can try?

Is there a way to remove a photo from photoprism so that it can be re-indexed?

**Reply by lastzero** [reply]
**The supported metadata fields are listed here: https://www.photoprism.app/kb/metadata**

### TakenAt

DateTimeOriginal, CreationDate, CreateDate, MediaCreateDate, ContentCreateDate, DateTimeDigitized, DateTime, SubSecDateTimeOriginal, SubSecCreateDate, DateCreated

### TakenAtLocal

DateTimeOriginal, CreationDate, CreateDate, MediaCreateDate, ContentCreateDate, DateTimeDigitized, DateTime, SubSecDateTimeOriginal, SubSecCreateDate

**Reply by jonfk** [reply]
Thanks for the quick replies, especially on the weekend!

I was able to figure out what was the issue. It seems that lightroom updated a few exif fields but not all of them. So my files ended up with fields as follows
```
exiftool -CreateDate -ModifyDate -DateCreated -DateTimeOriginal DSCN5043.JPG
Create Date                     : 2006:08:01 09:06:17
Modify Date                     : 2006:08:01 09:06:17
Date Created                    : 2018:09:16
Date/Time Original              : 2018:09:16 09:06:17
```

I used exiftool to update the fields that were set incorrectly using the following command:
`exiftool "-CreateDate<DateTimeOriginal" "-ModifyDate<DateTimeOriginal" DSCN5043.JPG`
which got me 
```
exiftool -CreateDate -ModifyDate -DateCreated -DateTimeOriginal -DateTimeCreated DSCN5043.JPG
Create Date                     : 2018:09:16 09:06:17
Modify Date                     : 2018:09:16 09:06:17
Date Created                    : 2018:09:16
Date/Time Original              : 2018:09:16 09:06:17
Date/Time Created               : 2018:09:16 09:06:17
```

The dates were properly adjusted in photoprism afterwards. 

Thank you for your help. I wouldn't have dug into the exact exif fields without your comment. I look forward to sponsoring you. The software has been pretty great. Just wish there were a few features already supported but it seems that you are working towards them already.
